src/main/kotlin/no/nav/sykdig/utils/FellesformatMapper.kt

```diff
@@ -0,0 +1,305 @@
+    (periode.fom..periode.tom).daysBetween().toInt()
+
+fun ClosedRange<LocalDate>.daysBetween(): Long = ChronoUnit.DAYS.between(start, endInclusive)
+
+fun tilArbeidsgiver(arbeidsgiver: Arbeidsgiver?): HelseOpplysningerArbeidsuforhet.Arbeidsgiver =
+    HelseOpplysningerArbeidsuforhet.Arbeidsgiver().apply {
+        harArbeidsgiver =
+            when (arbeidsgiver?.harArbeidsgiver) {
+                HarArbeidsgiver.EN_ARBEIDSGIVER -> CS().apply {
+                    dn = "Én arbeidsgiver"
+                    v = "1"
+                }
+                HarArbeidsgiver.FLERE_ARBEIDSGIVERE -> CS().apply {
+                    dn = "Flere arbeidsgivere"
+                    v = "2"
+                }
+                HarArbeidsgiver.INGEN_ARBEIDSGIVER -> CS().apply {
+                    dn = "Ingen arbeidsgiver"
+                    v = "3"
+                }
+                else -> {
+                    throw RuntimeException("Arbeidsgiver type er ukjent, skal ikke kunne skje")
+                }
+            }
+
+        navnArbeidsgiver = arbeidsgiver.navn
+        yrkesbetegnelse = arbeidsgiver.yrkesbetegnelse
+        stillingsprosent = arbeidsgiver.stillingsprosent
+    }
+
+fun tilMedisinskVurdering(
+    medisinskVurdering: MedisinskVurdering?,
+    skjermesForPasient: Boolean
+): HelseOpplysningerArbeidsuforhet.MedisinskVurdering {
+
+    val biDiagnoseListe: List<CV>? = medisinskVurdering?.biDiagnoser?.map {
+        toMedisinskVurderingDiagnode(it)
+    }
+
+    return HelseOpplysningerArbeidsuforhet.MedisinskVurdering().apply {
+        if (medisinskVurdering?.hovedDiagnose != null) {
+            hovedDiagnose = HelseOpplysningerArbeidsuforhet.MedisinskVurdering.HovedDiagnose().apply {
+                diagnosekode = toMedisinskVurderingDiagnode(medisinskVurdering.hovedDiagnose!!)
+            }
+        }
+        if (!biDiagnoseListe.isNullOrEmpty()) {
+            biDiagnoser = HelseOpplysningerArbeidsuforhet.MedisinskVurdering.BiDiagnoser().apply {
+                diagnosekode.addAll(biDiagnoseListe)
+            }
+        }
+        isSkjermesForPasient = skjermesForPasient
+        annenFraversArsak = medisinskVurdering?.annenFraversArsak?.let {
+            ArsakType().apply {
+                arsakskode.add(CS())
+                beskriv = medisinskVurdering.annenFraversArsak!!.beskrivelse
+            }
+        }
+        isSvangerskap = medisinskVurdering?.svangerskap
+        isYrkesskade = medisinskVurdering?.yrkesskade
+        yrkesskadeDato = medisinskVurdering?.yrkesskadeDato
+    }
+}
+
+fun toMedisinskVurderingDiagnode(diagnose: Diagnose): CV =
+    CV().apply {
+        s = diagnose.system
+        v = diagnose.kode
+        dn = diagnose.tekst
+    }
+
+enum class RestrictionCode(override val codeValue: String, override val text: String, override val oid: String = "2.16.578.1.12.4.1.1.8134") : Kodeverk {
+    RESTRICTED_FOR_EMPLOYER("A", "Informasjonen skal ikke vises arbeidsgiver")
```

Comment:
Må ikke denne samsvare med hva det heter i resten av verdikjeden..?

Reply:
jo, ser at eg var litt for rask her, jobber med å rette dette nå og lage litt tester